classifier/train.py

```diff
@@ -0,0 +1,696 @@
+                        the first dimension is the audio file, and the second
+                        dimension is the frame in the audio file. Evaluation
+                        will additionally be done at the audio file level
+                        (Type: bool)
+
+    Keyword Args:
+        verbose:  If True, print verbose messages
+                  (Type: bool)
+        batch_size: Number of smamples per batch
+                    (Type: int)
+        num_epochs: Number of training epochs
+                    (Type: int)
+        learning_rate: Learning rate value
+                       (Type: float)
+        weight_decay: L2 regularization factor
+                      (Type: float)
+    """
+    loss = 'categorical_crossentropy'
+    metrics = ['accuracy']
+    monitor = 'val_loss'
+    #set_random_seed(random_state)
+
+    # Set up data inputs
+    enc = OneHotEncoder(n_values=num_classes, sparse=False)
+
+    X_train = train_data['features']
+    y_train = enc.fit_transform(train_data['labels'].reshape(-1, 1))
+
+    if valid_data:
+        validation_data = (valid_data['features'],
+                           enc.fit_transform(valid_data['labels'].reshape(-1, 1)))
+        valid_split = 0.0
+    else:
+        validation_data = None
+
+    # Set up model
+    m, inp, out = construct_mlp_model(X_train.shape[1:],
+                                      weight_decay=weight_decay,
+                                      num_classes=num_classes)
+
+    # Set up callbacks
+    cb = []
+    weight_path = os.path.join(model_dir, 'model.h5')
+    cb.append(keras.callbacks.ModelCheckpoint(weight_path,
+                                              save_weights_only=True,
+                                              save_best_only=True,
+                                              monitor=monitor))
+    cb.append(keras.callbacks.EarlyStopping(monitor='val_loss', patience=patience))
+    history_checkpoint = os.path.join(model_dir, 'history_checkpoint.pkl')
+    cb.append(LossHistory(history_checkpoint))
+    history_csvlog = os.path.join(model_dir, 'history_csvlog.csv')
+    cb.append(keras.callbacks.CSVLogger(history_csvlog, append=True,
+                                        separator=','))
+    metric_cb = MetricCallback(valid_data, verbose=verbose)
+    cb.append(metric_cb)
+
+    # Fit model
+    LOGGER.debug('Compiling model...')
+    m.compile(Adam(lr=learning_rate), loss=loss, metrics=metrics)
+    LOGGER.debug('Fitting model to data...')
+    m.fit(x=X_train, y=y_train, batch_size=batch_size, epochs=num_epochs,
+          validation_data=validation_data, validation_split=valid_split,
+          callbacks=cb, verbose=2)
+
+    # Compute metrics for train and valid
+    train_pred = m.predict(X_train)
+    train_metrics = compute_metrics(y_train, train_pred, num_classes=num_classes)
+    # Set up train and validation metrics
+    train_metrics = {
+        'loss': metric_cb.train_loss[-1],
```

Comment:
Also, we need to make sure we're reloading the checkpoint with the best loss, since Keras just leaves the model in whatever state the model was when finished training